### PR TITLE
LTP: Update ltp-open-posix to run specific group test

### PR DESCRIPTION
**Pull request by Lakshmipathi**
Run specific group test from ltp-open-posix. Parse the
results and send it to lava-test-case.

Signed-off-by: lakshmipathi ganapathi <lakshmipathi.ganapathi@collabora.com>

### CODE REVIEW

**Comment by gctucker** [comment]
Here's some sample results from the open POSIX timers tests using this patch:
https://linux.kernelci.org/test/plan/id/60e2ae83e8c2492bdc11796e/ (x86 AMD)
https://linux.kernelci.org/test/plan/id/60e2b414ad077537cc11796a/ (arm)

**Review by nareshkamboju — APPROVED** [review]

File: automated/linux/ltp-open-posix/ltp-open-posix.sh
Comment (nareshkamboju) [inline automated/linux/ltp-open-posix/ltp-open-posix.sh:71]: LTP open posix test suites results verdicts / Labels are different than pass, fail and skip.
Please consider below labels also.
SKIPPED|: UNSUPPORTED|: UNTESTED|: UNRESOLVED|: HUNG

Comment (Lakshmipathi) [reply automated/linux/ltp-open-posix/ltp-open-posix.sh:71]: Okay. Let me add these labels too.

**Comment by nareshkamboju** [comment]
@Lakshmipathi Please fix shellcheck warnings / errors.

**Comment by Lakshmipathi** [comment]
@nareshkamboju , sure just now update the patch to resolve spellcheck issue. Will look into labels.

File: automated/linux/ltp-open-posix/ltp-open-posix.sh
Comment (Lakshmipathi) [reply automated/linux/ltp-open-posix/ltp-open-posix.sh]: Handled these labels and tested the changes with this job: https://lava.collabora.co.uk/scheduler/job/4156348

**Comment by gctucker** [comment]
@nareshkamboju It looks like you or some maintainer needs to give approval for the GitHub checks to run, for first-time contributions.

**Comment by Lakshmipathi** [comment]
Updated the patch to resolve recent linting errors

File: automated/linux/ltp-open-posix/ltp-open-posix.yaml
Comment (nareshkamboju) [inline automated/linux/ltp-open-posix/ltp-open-posix.yaml:28]: @Lakshmipathi Please describe the purpose of "GRP_TEST" in params: section.

**Comment by Lakshmipathi** [comment]
@nareshkamboju could you please trigger linting check? This time it should pass.

**Comment by bhcopeland** [comment]
@Lakshmipathi triggered

**Comment by Lakshmipathi** [comment]
thanks @bhcopeland

**Review by nareshkamboju — APPROVED** [review]

File: automated/linux/ltp-open-posix/ltp-open-posix.yaml
Comment (nareshkamboju) [inline automated/linux/ltp-open-posix/ltp-open-posix.yaml:26]: Add an extra empty line

File: automated/linux/ltp-open-posix/ltp-open-posix.sh
Comment (nareshkamboju) [inline automated/linux/ltp-open-posix/ltp-open-posix.sh:95]: Is this empty line needed here ?

File: automated/linux/ltp-open-posix/ltp-open-posix.sh
Comment (nareshkamboju) [inline automated/linux/ltp-open-posix/ltp-open-posix.sh:106]: you may delete this empty line.

File: automated/linux/ltp-open-posix/ltp-open-posix.sh
Comment (nareshkamboju) [inline automated/linux/ltp-open-posix/ltp-open-posix.sh:82]: Please share the latest LAVA test log links with this later code.

**Review by nareshkamboju — CHANGES_REQUESTED** [review]
Please check inline comments.

File: automated/linux/ltp-open-posix/ltp-open-posix.sh
Comment (Lakshmipathi) [reply automated/linux/ltp-open-posix/ltp-open-posix.sh:82]: ok, here is the LAVA job with these changes: https://lava.collabora.co.uk/scheduler/job/4189498

**Comment by Lakshmipathi** [comment]
Updated PR and resolved latest review comments.

**Comment by nareshkamboju** [comment]
+1
Looks good to me.

**Review by nareshkamboju — APPROVED** [review]

File: automated/linux/ltp-open-posix/ltp-open-posix.sh
Comment (roxell) [inline automated/linux/ltp-open-posix/ltp-open-posix.sh:77]: Can we squash these two lines into one?
something like this:
| sed 's/PASS/pass/; s/FAIL/fail/; s/CONF/skip/'|awk '{ if($5 == 1) {print $1 " " $4} else {print $1 " " $2} }'

Comment (Lakshmipathi) [reply automated/linux/ltp-open-posix/ltp-open-posix.sh]: @roxell thanks. Updated the PR to use single line and tested the change. https://lava.collabora.co.uk/scheduler/job/4202810

**Comment by Lakshmipathi** [comment]
Could someone please trigger shellcheck for recent change?

**Pull request merged**
